### ISSUE TITLE
Unable to execute HTTP request: company-customer-stage.s3.cn-northwest-1.amazonaws.com

### ISSUE THREAD

**Issue by xxyqz** [snowflakedb/snowflake-kafka-connector#879]
I used Kafka 3.2.1 and Snowflake-Kafka-connector 2.2.2. In public internet, the connection is OK for US region, but failed for China region.

```
[2024-07-11 13:58:04,497] INFO [company|task-1] [SF_KAFKA_CONNECTOR] pipe SNOWFLAKE_KAFKA_CONNECTOR_company_PIPE_sf_table_1_0: cleaner started (com.snowflake.kafka.connector.internal.SnowflakeSinkServiceV1:46)
Jul 11, 2024 1:59:05 PM net.snowflake.client.jdbc.SnowflakeFileTransferAgent uploadWithoutConnection
SEVERE: Exception encountered during file upload in uploadWithoutConnection
net.snowflake.client.jdbc.SnowflakeSQLLoggedException: AWS operation failed: Operation=upload, Error message=Unable to execute HTTP request: company-ng-ds1-customer-stage.s3.cn-northwest-1.amazonaws.com
        at net.snowflake.client.jdbc.cloud.storage.SnowflakeS3Client.handleS3Exception(SnowflakeS3Client.java:767)
        at net.snowflake.client.jdbc.cloud.storage.SnowflakeS3Client.upload(SnowflakeS3Client.java:563)
        at net.snowflake.client.jdbc.SnowflakeFileTransferAgent.pushFileToRemoteStore(SnowflakeFileTransferAgent.java:2080)
        at net.snowflake.client.jdbc.SnowflakeFileTransferAgent.uploadWithoutConnection(SnowflakeFileTransferAgent.java:2199)
        at com.snowflake.kafka.connector.internal.SnowflakeInternalStage.putWithCache(SnowflakeInternalStage.java:176)
        at com.snowflake.kafka.connector.internal.SnowflakeConnectionServiceV1.lambda$putWithCache$1(SnowflakeConnectionServiceV1.java:855)
        at com.snowflake.kafka.connector.internal.InternalUtils.backoffAndRetry(InternalUtils.java:373)
        at com.snowflake.kafka.connector.internal.SnowflakeConnectionServiceV1.putWithCache(SnowflakeConnectionServiceV1.java:851)
        at com.snowflake.kafka.connector.internal.SnowflakeSinkServiceV1$ServiceContext.flush(SnowflakeSinkServiceV1.java:844)
        at com.snowflake.kafka.connector.internal.SnowflakeSinkServiceV1$ServiceContext.flushBuffer(SnowflakeSinkServiceV1.java:773)
        at com.snowflake.kafka.connector.internal.SnowflakeSinkServiceV1$ServiceContext.access$200(SnowflakeSinkServiceV1.java:373)
        at com.snowflake.kafka.connector.internal.SnowflakeSinkServiceV1.insert(SnowflakeSinkServiceV1.java:159)
        at com.snowflake.kafka.connector.SnowflakeSinkTask.put(SnowflakeSinkTask.java:304)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:581)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:333)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
        at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:188)
        at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: net.snowflake.client.jdbc.internal.amazonaws.SdkClientException: Unable to execute HTTP request: company-ng-ds1-customer-stage.s3.cn-northwest-1.amazonaws.com
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.handleRetryableException(AmazonHttpClient.java:1219)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1165)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:814)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:781)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:755)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:715)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:697)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:561)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:541)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5519)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5466)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.access$300(AmazonS3Client.java:422)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client$PutObjectStrategy.invokeServiceCall(AmazonS3Client.java:6600)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.uploadObject(AmazonS3Client.java:1891)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1851)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.transfer.internal.UploadCallable.uploadInOneChunk(UploadCallable.java:169)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.transfer.internal.UploadCallable.call(UploadCallable.java:149)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.transfer.internal.UploadMonitor.call(UploadMonitor.java:115)
        at net.snowflake.client.jdbc.internal.amazonaws.services.s3.transfer.internal.UploadMonitor.call(UploadMonitor.java:45)
        ... 4 more
Caused by: java.net.UnknownHostException: company-ng-ds1-customer-stage.s3.cn-northwest-1.amazonaws.com
        at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797)
        at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1533)
        at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1386)
        at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1307)
        at net.snowflake.client.jdbc.internal.amazonaws.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:27)
        at net.snowflake.client.jdbc.internal.amazonaws.http.DelegatingDnsResolver.resolve(DelegatingDnsResolver.java:38)
        at net.snowflake.client.jdbc.internal.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:112)
        at net.snowflake.client.jdbc.internal.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
        at jdk.internal.reflect.GeneratedMethodAccessor21.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at net.snowflake.client.jdbc.internal.amazonaws.http.conn.ClientConnectionManagerFactory$Handler.invoke(ClientConnectionManagerFactory.java:76)
        at net.snowflake.client.jdbc.internal.amazonaws.http.conn.$Proxy42.connect(Unknown Source)
        at net.snowflake.client.jdbc.internal.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
        at net.snowflake.client.jdbc.internal.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
        at net.snowflake.client.jdbc.internal.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
        at net.snowflake.client.jdbc.internal.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
        at net.snowflake.client.jdbc.internal.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at net.snowflake.client.jdbc.internal.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
        at net.snowflake.client.jdbc.internal.amazonaws.http.apache.client.impl.SdkHttpClient.execute(SdkHttpClient.java:72)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1346)
        at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1157)
        ... 21 more
```

I checked my account's allowlist by using  system$allowlist(): 
```
`select system$allowlist()`
{"host":"company-customer-stage.s3.cn-northwest-1.amazonaws.com.cn","port":443,"type":"STAGE"}
```
The result looks like correct, I don't know why Kafka connector got "amazonaws.com" without ".cn" ?

**Comment by sfc-gh-wfateem** [comment]
Hi @xxyqz,

Thanks for bringing this up. This is an issue with the JDBC driver, and the PR https://github.com/snowflakedb/snowflake-jdbc/pull/1820 will address it.

Once that fix is merged into the JDBC main branch, the Kafka Connector will need to update the JDBC driver bundled with it.

**Comment by xxyqz** [comment]
Got it. Thank you very much.
